kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -131,6 +148,16 @@ private void validate(Engine engine, SnapshotImpl snapshot, boolean isNewTable)
             SchemaUtils.validatePartitionColumns(
                     schema.get(), partitionColumns.orElse(Collections.emptyList()));
         }
+
+        transactionId.ifPresent(txnId -> {
+            Optional<Long> lastTxnVersion = snapshot.getLatestTransactionVersion(txnId.getAppId());
```

Comment:
will the conflict checker check this during logical conflicts?

Reply:
Yep, see my prev comment resp [here](url).